### DISCUSSION TITLE
podman not able to connect to ports when starting oracle db container

### DISCUSSION THREAD

**Discussion by abhinaya-raja** [containers/podman#23025] [Q&A]
podman not able to connect to ports when starting oracle db container.
echo "Podman run container"
podman run -p 1521:1521 -p 5500:5500 --entrypoint "/bin/bash" -t -d --name analytics-db --user oracle --cpus=2 --memory="4096" --restart=always --expose=1500-5600 --publish-all -e ORACLE_SID=ANADB -e ORACLE_PDB=ANAPDB -e ORACLE_PWD=${SYSDBA_PASS} -e  ORACLE_CHARACTERSET=AL32UTF8 -v analytics-db-data:/opt/oracle/oradata xxx
podman ps -a

output:
22:53:40  CONTAINER ID  IMAGE                                                        COMMAND     CREATED                 STATUS                            PORTS       NAMES
22:53:40  59b64f9ee611  xxx             Less than a second ago  Up Less than a second (starting)              analytics-db

container configuration:
[containers]
netns="host"
userns="host"
ipcns="host"
utsns="host"
cgroupns="host"
cgroups="enabled"
cgroup_enable="memory" 
systemd.unified_cgroup_hierarchy=0
cgroup_memory=1
legacyCgroupv1=true
swapaccount=1
log_driver = "k8s-file"
volumes = [
	"/proc:/proc"
]
[engine]
cgroup_manager = "cgroupfs"
events_logger="file"
runtime="crun"

podman inspect analytics-db command shows ports are empty

 "HostConfig": {
11:57:54                 "Binds": [
11:57:54                      "analytics-db-data:/opt/oracle/oradata:rw,rprivate,nosuid,nodev,rbind"
11:57:54                 ],
11:57:54                 "CgroupManager": "cgroupfs",
11:57:54                 "CgroupMode": "host",
11:57:54                 "ContainerIDFile": "",
11:57:54                 "LogConfig": {
11:57:54                      "Type": "k8s-file",
11:57:54                      "Config": null,
11:57:54                      "Path": "/var/lib/containers/storage/overlay-containers/e7ba936621c5789fcc982439352f2dcf5e3e988b265c4d5866dc4a6de4ccdf36/userdata/ctr.log",
11:57:54                      "Tag": "",
11:57:54                      "Size": "0B"
11:57:54                 },
11:57:54                 "NetworkMode": "host",
11:57:54                 "PortBindings": {},

**Comment by Luap99** [comment]
>  netns="host"

When you set the netns to host all port mappings are ignored as you already share the host network namespace so there is no need to forward ports.

**Reply by abhinaya-raja** [reply]
I tried removing netns=“host” option, now my container is able to connect to ports
10:25:47  CONTAINER ID  IMAGE                                                                  COMMAND               CREATED             STATUS      PORTS                                           NAMES
10:25:47  7ebeb7bfb6f2  xxx                                About a minute ago  Created     0.0.0.0:1521->1521/tcp, 0.0.0.0:5500->5500/tcp  analytics-db

But I see this error - container is in created state but not running
10:25:47  Healthcheck container
10:25:47  Error: container is not running
10:25:47  Sys DBA File copied
10:25:47  Error: can only create exec sessions on running containers: container state improper

Error: failed to mount runtime directory for rootless netns: no such file or directory

23:40:11  Error: failed to start rootless-netns slirp4netns: /usr/bin/slirp4netns failed: "open(\"/dev/net/tun\"): No such file or directory\nWARNING: Support for seccomp is experimental\nWARNING: Support for IPv6 is experimental\nchild failed(1)\nWARNING: Support for seccomp is experimental\nWARNING: Support for IPv6 is experimental\n"

**Reply by Luap99** [reply]
If you run nested containers you have to give them enough permissions to do things like networking. That why host is the default because it just works without extra configuration.

Generally the easiest is to start the outer container with --privileged

**Reply by abhinaya-raja** [reply]
After adding --privileged and network_cmd_options = [] in service.conf under [engine], ran the pipeline, not seeing the mount failed error.
But container is in created state not able to run.
podman container logs shows below error:

"State": {
09:50:47                 "OciVersion": "1.1.0-rc.3",
09:50:47                 "Status": "created",
09:50:47                 "Running": false,
09:50:47                 "Paused": false,
09:50:47                 "Restarting": false,
09:50:47                 "OOMKilled": false,
09:50:47                 "Dead": false,
09:50:47                 "Pid": 0,
09:50:47                 "ExitCode": 0,
09:50:47                 "Error": "failed to start rootless-netns slirp4netns: /usr/bin/slirp4netns failed: \"open(\\\"/dev/net/tun\\\"): No such file or directory\\nWARNING: Support for seccomp is experimental\\nWARNING: Support for IPv6 is experimental\\nchild failed(1)\\nWARNING: Support for seccomp is experimental\\nWARNING: Support for IPv6 is experimental\\n\"",
09:50:47                 "StartedAt": "0001-01-01T00:00:00Z",
09:50:47                 "FinishedAt": "0001-01-01T00:00:00Z",
09:50:47                 "Health": {
09:50:47                      "Status": "",
09:50:47                      "FailingStreak": 0,
09:50:47                      "Log": null
09:50:47                 },
09:50:47                 "CheckpointedAt": "0001-01-01T00:00:00Z",
09:50:47                 "RestoredAt": "0001-01-01T00:00:00Z"
09:50:47            },

**Reply by Luap99** [reply]
I know nothing about your environment but it sounds like you are trying to run nested containers so you must set --privileged on the outer container, maybe this can help: https://www.redhat.com/sysadmin/podman-inside-container
